src/layout.cpp

```diff
@@ -122,7 +122,14 @@ LayoutNavEntry *LayoutNavEntry::find(LayoutNavEntry::Kind kind,
   return result;
 }
 
-
+void LayoutNavEntry::adjustVisibility()
```

Comment:
I would expect that instead of fixing the visibility after processing, it can also be set while parsing by changing

```c++
bool isVisible = elemIsVisible(attrib);
```

by e.g.

```c++
bool isVisible = elemIsVisible(attrib) && (parent()==0 || parent()->visible());
```

at various places. Did you try or consider this?

Reply:
No I didn't consider this, but I think that in that case also the recursive part is missing i.e. checking grandparent etc.